### PR TITLE
Add fwupd probe test and some fixes

### PR DESCRIPTION
**Pull request by evgenyz**
The probe won't check the schema version anymore and would react to edge cases in a more consistent way.

This PR will also add D-Bus mock test infrastructure.

Detailed description of changes is in individual commit messages.

### CODE REVIEW

**Comment by evgenyz** [comment]
I've managed to make the mock DBus module work in all environments we care about ATM. After merging #1954 (and rebasing on it) CI should be green.

In the mean time feedback on the changes is welcome.

File: tests/probes/fwupdsecattr/test_probes_fwupdsecattr_mock.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    $OSCAP oval eval --results $RF $DF
+    clean_dbus_mock $DBUS_MOCK_NAME
+
+    if [ -f $RF ]; then
+        verify_results "def" $DF $RF 1 && verify_results "tst" $DF $RF 1
+        ret_val=$?
+    else
+        ret_val=1
+    fi
+
```
Comment (jan-cerny) [inline tests/probes/fwupdsecattr/test_probes_fwupdsecattr_mock.sh:28]: If the warnings like `W: oscap:     HSI key not found: org.fwupd.hsi.Kernel.InvalidOrNonExisting` are expected here, there should be an assert that tests them.

File: tests/probes/fwupdsecattr/test_probes_fwupdsecattr.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+. $builddir/tests/test_common.sh
+
+set -e -o pipefail
+
+function test_probes_fwupdsecattr {
```
[the remark this answers is not in the document]
Comment (evgenyz) [reply tests/probes/fwupdsecattr/test_probes_fwupdsecattr.sh:7]: Fixed.

File: tests/probes/fwupdsecattr/test_probes_fwupdsecattr_mock.sh (hunk 1 above)
Comment (evgenyz) [reply tests/probes/fwupdsecattr/test_probes_fwupdsecattr_mock.sh:30]: Done.

**Comment by jan-cerny** [comment]
@evgenyz Why has the test_probes_fwupdsecattr.sh test been skipped in the Fedora Latest container action?

**Comment by evgenyz** [comment]
> @evgenyz Why has the test_probes_fwupdsecattr.sh test been skipped in the Fedora Latest container action?

The `test_probes_fwupdsecattr.sh` test tries to use real `fwupd` service, but it does not support virtualized/container environments. So, when it receives 'not supported' it bails.

**Review by jan-cerny — APPROVED** [review]
I have also run the tests locally because the first one is skip in the containerized CI environment.

```
[jcerny@thinkpad build{pr/1944}]$ ctest --verbose -R fwupdsecattr
UpdateCTestConfiguration  from :/home/jcerny/work/git/openscap/build/DartConfiguration.tcl
UpdateCTestConfiguration  from :/home/jcerny/work/git/openscap/build/DartConfiguration.tcl
Test project /home/jcerny/work/git/openscap/build
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 198
    Start 198: probes/fwupdsecattr/test_probes_fwupdsecattr.sh

198: Test command: /home/jcerny/work/git/openscap/tests/probes/fwupdsecattr/test_probes_fwupdsecattr.sh
198: Working Directory: /home/jcerny/work/git/openscap/build/tests/probes/fwupdsecattr
198: Environment variables: 
198:  srcdir=/home/jcerny/work/git/openscap/tests/probes/fwupdsecattr
198:  top_srcdir=/home/jcerny/work/git/openscap
198:  builddir=/home/jcerny/work/git/openscap/build
198: Test timeout computed to be: 10000000
198: Definition oval:0:def:1: true
198: Evaluation done.
1/2 Test #198: probes/fwupdsecattr/test_probes_fwupdsecattr.sh ........   Passed    1.77 sec
test 199
    Start 199: probes/fwupdsecattr/test_probes_fwupdsecattr_mock.sh

199: Test command: /home/jcerny/work/git/openscap/tests/probes/fwupdsecattr/test_probes_fwupdsecattr_mock.sh
199: Working Directory: /home/jcerny/work/git/openscap/build/tests/probes/fwupdsecattr
199: Environment variables: 
199:  srcdir=/home/jcerny/work/git/openscap/tests/probes/fwupdsecattr
199:  top_srcdir=/home/jcerny/work/git/openscap
199:  builddir=/home/jcerny/work/git/openscap/build
199: Test timeout computed to be: 10000000
199: stderr file: test_probes_fwupdsecattr_mock.err.3PKCsE
199: Initializing D-Bus mock from template: org.freedesktop.fwupd.py
199: 1678784456.315 GetHostSecurityAttrs
199: Definition oval:0:def:2: unknown
199: Definition oval:0:def:1: true
199: Evaluation done.
199: Shutting down D-Bus mock: org.freedesktop.fwupd
199: 1678784456.428 Exit
2/2 Test #199: probes/fwupdsecattr/test_probes_fwupdsecattr_mock.sh ...   Passed    0.47 sec

The following tests passed:
	probes/fwupdsecattr/test_probes_fwupdsecattr.sh
	probes/fwupdsecattr/test_probes_fwupdsecattr_mock.sh

100% tests passed, 0 tests failed out of 2

Total Test time (real) =   2.25 sec
[jcerny@thinkpad build{pr/1944}]$ 
```
